src/common/observablevector.ts

```diff
@@ -322,6 +330,11 @@ class ObservableVector<T> extends Vector<T> implements IObservableVector<T> {
    */
   set(index: number, value: T): void {
     let oldValues = [this.at(index)];
+    // Bail if the value does not change.
+    let itemCmp = this._itemCmp;
+    if (itemCmp(oldValues[0], value)) {
```

Comment:
But in this case, I'm with `ian-r-rose` I think `itemCmp` should be used in remove, because this should work:

```
let thing1 = someThing;
let thing2 = someThingEqualToThing1;
vector.set(0, thing1);
vector.set(0, thing2);
vector.remove(thing2); // thing1 still at 0
```